mcda/utils/utils_for_main.py

```diff
@@ -322,10 +361,15 @@ def read_matrix(input_matrix_path: str) -> pd.DataFrame:
     :param input_matrix_path: str
     :rtype: pd.DataFrame
     """
+    try:
+        ensure_directory_exists(input_directory_path)
+    except IOError:
```

Comment:
I don't think that here you will get any exception to catch because firstly, you already implemented try catch in `ensure_directory_exists` method and secondly, in `ensure_directory_exists` method you are catching `Exception` the parent class of all exceptions and here you are catching sub-class of `Exception`. 
Possible solutions:
- either throw all the possible exceptions from `ensure_directory_exists` and catch them here.
  - this approach will give you more control over the log messages and it would be easier to trace back to the error
- remove try except from here and just log the errors in `ensure_directory_exists` but this way you lose control over the messages (like from where this method was called).

Reply:
That is my new logic:

1. in `ensure_directory_exists` I catch any exceptions, log in the message and raise the exception again to propagate it to the caller functions
2. in the caller functions I catch the Exception and log in information about where exactly the problem happened
3. additionally, I have added an extra try-except for the writing/dumping of the different objects.

How does it look now to you?

Reply:
If you want then I think in step 1 we can raise a specific exception instead of just raising the base class Exception e.

Something like: https://stackoverflow.com/a/2052396

Something like:
```
raise OsError(..... some text with e.message) #  not sure what could be the appropriate exception, this you will get to know in the stack traces when you run the application. 
```

Reply:
OK, I'm going to use `FileNotFoundError` exception.